### DISCUSSION TITLE
Question on how to plot GOS map of a single phase for a multi-phase EBSD data

### DISCUSSION THREAD

**Discussion by bjzhou96** [mtex-toolbox/mtex#1297] [Ask Anything]
Dear overyone
I have trouble with plotting GOS map of only a single phase for a multi-phase EBSD data. 
[grains,ebsd.grainId] = calcGrains(ebsd);
ebsd(grains(grains.grainSize<3)) = [];
[grains,ebsd.grainId] = calcGrains(ebsd,'threshold',[2*degree, 10*degree]);

% lets smooth the grain boundaries a bit
grains = smooth(grains,5);
grod = ebsd('Amphibole').calcGROD(grains)
GOS = grainMean(ebsd('Amphibole'), grod.angle);

plot(grains('Amphibole'), GOS ./ degree)
set(gca,'clim',[0 5]) 
mtexColorbar('title','GOS in degree')

The error is :
Error using grain2d/plot (line 61)
Number of grains must be the same as the number of data

Error in ALLmaps (line 48)
plot(grains('Amphibole'), GOS ./ degree)

I think I forgot to add "('Amphibole')" somewhere, but I still didn't make it after many attempts. Anyone know how to fix it, please help me. Thank you in advance!
 
Best wishes
Baojun Zhou

**Comment by kilir** [comment]
Hi,
in general, try:
```
grod = ebsd('Am').calcGROD(grains);
GOS = grainMean(ebsd('Am'), grod.angle, grains('Am'));
plot(grains('Am'), GOS ./ degree)
```
If you are only interested in the GOS, there's also simply  `grains.GOS`.

Hope that helps.

Cheers,
Rüdiger

**Comment by bjzhou96** [comment]
Hello Rüdiger

Thank you for your quick reply! It is very kind of you to help me. It works, again！

Best wishes
Baojun Zhou
